tools/lib/esbuildConfig.js

```diff
@@ -69,16 +64,15 @@ const esbuildConfig = ({ nodeEnv, name }) => {
 		outfile,
 		bundle: true,
 		sourcemap: true,
-		jsx: "automatic",
-		jsxDev: nodeEnv === "development",
+		inject: ["tools/lib/react-shim.mjs"],
 		define: {
 			"process.env.NODE_ENV": JSON.stringify(nodeEnv),
 			"process.env.SPORT": JSON.stringify(sport),
 		},
 		plugins: [pluginSportFunctions(nodeEnv)],
 
-		// This is needed because dropbox conditionally requries various node builtins, and esbuild chokes on that even though it never actually makes it to the browser
-		external: name === "ui" ? ["crypto", "util"] : undefined,
+		// This is needed because dropbox conditionally requries various node builtins, and esbuild chokes on that even though it never actually makes it to the browser. Skip it for the worker though, otherwise that introduces a spurious error because the type export in worker/index.ts results in a module.exports being added.
+		platform: name === "ui" ? "node" : undefined,
```

Comment:
I never had this problem, I am skeptical this change is needed. But the line you deleted is needed, otherwise the build fails for the same reason I wrote in the comment.